cpp/src/cider/tests/nextgen/compiler/ArrayTest.cpp

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <google/protobuf/util/json_util.h>
+#include <gtest/gtest.h>
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+#include "exec/nextgen/Nextgen.h"
+#include "exec/nextgen/context/Batch.h"
+#include "exec/plan/parser/SubstraitToRelAlgExecutionUnit.h"
+#include "exec/plan/parser/TypeUtils.h"
+
+#include "tests/TestHelpers.h"
+#include "tests/utils/ArrowArrayBuilder.h"
+#include "tests/utils/Utils.h"
+#include "util/Logger.h"
+
+using namespace cider::exec::nextgen;
+
+static const std::shared_ptr<CiderAllocator> allocator =
+    std::make_shared<CiderDefaultAllocator>();
+
+operators::TranslatorPtr initSqlToTranslators(const std::string& sql,
+                                              const std::string& create_ddl) {
+  // SQL Parsing
+  auto json = RunIsthmus::processSql(sql, create_ddl);
+  ::substrait::Plan plan;
+  google::protobuf::util::JsonStringToMessage(json, &plan);
+
+  generator::SubstraitToRelAlgExecutionUnit substrait2eu(plan);
+  auto eu = substrait2eu.createRelAlgExecutionUnit();
+
+  // Pipeline Building
+  auto pipeline = parsers::toOpPipeline(eu);
+  transformer::Transformer transformer;
+  return transformer.toTranslator(pipeline);
+}
+
+context::RuntimeCtxPtr executeAndReturnRuntimeCtx(const std::string& create_ddl,
```

Comment:
why not return outBatch directly?

Reply:
outBatch will be released when runtimeContext is released.